File: python/decorator/StarbuzzCoffee/decorator/condiments/Soy.py
from component.Beverage import Beverage
from decorator.CondimentDecorator import CondimentDecorator


class Soy(CondimentDecorator):
    __beverage = Beverage()

    def __init__(self, beverage):
        super().__init__()
        self.__beverage = beverage

    def get_description(self):
        return self.__beverage.get_description() + ", Soy"

    def cost(self):
        return self.__beverage.cost() + 0.10

File: ruby/decorator/StarbuzzCoffee/decorator/condiments/soy.rb
require './decorator/condiment_decorator'

class Soy < CondimentDecorator

  def initialize(beverage)
    @@beverage = beverage
  end

  def get_description
    super
    @@beverage.get_description + ", Soy"
  end

  def cost
    super
    @@beverage.cost + 0.22
  end

  private
  @@beverage = Beverage.new

end